src/EventStore.Core/Services/ElectionsService.cs

```diff
@@ -423,14 +456,16 @@ public class ElectionsService : IHandle<SystemMessage.BecomeShuttingDown>,
 
 		public static LeaderCandidate GetBestLeaderCandidate(Dictionary<Guid, ElectionMessage.PrepareOk> received,
 			MemberInfo[] servers, Guid? resigningLeaderInstanceId, int lastAttemptedView) {
+			
 			var best = received.Values
-				.OrderByDescending(x => x.EpochNumber)
-				.ThenByDescending(x => x.LastCommitPosition)
-				.ThenByDescending(x => x.WriterCheckpoint)
-				.ThenByDescending(x => x.ChaserCheckpoint)
-				.ThenByDescending(x => x.NodePriority)
-				.ThenByDescending(x => x.ServerId)
+				.OrderByDescending(x => x.EpochNumber) // latest election
+				.ThenByDescending(x => x.WriterCheckpoint) // log length (if not re-electing leader pick the most complete replica)
+				.ThenByDescending(x => x.ChaserCheckpoint) // in memory index position
+				.ThenByDescending(x => x.LastCommitPosition) // on disk index position
```

Comment:
the `LastCommitPosition` is not stored on disk - it is based on the index checkpoint which is updated only after each PTable dump. on startup, the index checkpoint is rebuilt up to the chaser checkpoint so it may be initialized to a value pointing to uncommitted data.

Reply:
We do not Accept masters without the correct epoch and data, this selection here is just for picking the node with the least work to become and active master.